src/game/room.c

```diff
@@ -320,9 +320,9 @@ int16_t Room_GetCeiling(FLOOR_INFO *floor, int32_t x, int32_t y, int32_t z)
 
     if (f->index) {
         data = &g_FloorData[f->index];
-        type = *data++ & DATA_TYPE;
+        type = *data & DATA_TYPE;
 
-        if (type == FT_TILT) {
+        if (!(*data++ & END_BIT) && type == FT_TILT) {
```

Comment:
Let's try to avoid pointer manipulation inside conditions and extract this to a variable.
BTW shouldn't this `END_BIT` check also apply to `FT_ROOF`?

Reply:
We don't need to worry about the roof as it doesn't look for any more entry types after that. End bit means no more functions for this sector, but there can still be value data. So it reads the slant values and does what it needs to with those. There is a loop further down which checks for objects changing the height, trapdoors and suchlike, but that starts again at the beginning of the sector's FD and the loop condition explicitly checks for the end bit.